rfd/9999-aws-api-integration-using-oidc.md

```diff
@@ -0,0 +1,554 @@
+## How
+
+AWS allows the [set up](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html) of an IdP using OIDC providers.
+Each provider has a set of roles, which limit their access.
+
+Turning Teleport into an OIDC IdP, and asking the user to create a new OIDC IdP that consumes the Teleport OIDC endpoints, allows us to easily call AWS API endpoints to, for instance, list RDS instances and their details.
+
+When configuring the provider, we'll need an AWS Role which Teleport uses to issue API Calls.
+
+To store the configuration above, we'll create a new resource Kind: `Integration`.
+It'll leverage the `subkind` prop to distinguish future integrations.
+
+### High Level Flow
+Simplified flow of interactions between User, Teleport and AWS:
+
+```
+                                                               ┌─────────┐
+                                                               │         │
+                ┌─────────────────────────────────────────────►│  AWS    │
+               2│                                              │         │
+                │                                              └──┬───▲──┘
+User────────────┤                                                 │   │
+                │       ┌───────────────────────┐                 │   │
+               1│       │       Teleport        │                 │   │
+                │       │                       │                 │   │
+               5│    ┌──┼──────┐       ┌────────┼──┐              │   │
+                └────►  │ Web  │       │ OIDC   │  │       3      │   │
+                     └──┼──┬───┘       │ IdP    │  │◄─────────────┘   │
+                        │  │           └┬───────┼──┘                  │
+                        │  │            │       │                     │
+                        │  │           4│       │                     │
+                        │  │            │       │                     │
+                        │  │  ┌─────────▼┐      │                     │
+                        │  │  │ CA KeySet│      │                     │
+                        │  │  │ +OIDC CA │      │                     │
+                        │  │  └─────────▲┘      │                     │
+                        │  │6           │7      │                     │
+                        │  │     ┌──────┴────┐  │                     │
+                        │  │     │ Discover  │  │           8         │
+                        │  └────►│ Service   ├──┼─────────────────────┘
+                        │        │ [ RDS ]   │  │
+                        │        └───────────┘  │
+                        │                       │
+                        └───────────────────────┘
+```
+
+#### User's Point of View
+When the user adds a new resource, using the Discover Wizard, they'll be asked where the resource lives.
+Choosing an RDS DB, the user is presented with the following flow:
```

Comment:
The "Add AWS integration" flow can be gone through separately from the Discover wizard though too, right? And it will be skipped if you already have an integration setup? So I'd probably rephrase this to something like "When a users adds a new AWS integration, they will go through the following. Users will also be asked to set up an AWS integration when adding RDS databases in Discover flow."